### PR TITLE
more fixes for llvm backend

### PR DESCRIPTION
**Pull request by dwightguth**
This is all the backwards-compatible fixes to get the llvm backend working with the iele mantis server.

### CODE REVIEW

File: data.md [hunk 1]
```diff
@@ -530,6 +530,8 @@ Decoding
 --------
 
 -   `#loadLen` and `#loadOffset` decode a `WordStack` into a single string in an RLP-like encoding which does not allow lists in its structure.
+-   `#rlpDecode` RLP decodes a single `String` into a `JSON`.
+-   `#rlpDecodeList` RLP decodes a single `String` into a `JSONList`, interpereting the string as the RLP encoding of a list.
```
Comment (bmmoore) [inline data.md]: Where are these being used?

File: iele-gas.md [hunk 2]
```diff
@@ -374,17 +374,21 @@ which is maintained by the next rules.
          <currentMemory> CURR => CURR +Int maxInt(0, NEWSIZE -Int bytesInWords((lengthBytes(LM)))) </currentMemory>
          <peakMemory> PEAK => maxInt(PEAK, CURR +Int maxInt(0, NEWSIZE -Int bytesInWords((lengthBytes(LM))))) </peakMemory>
 
-    rule <k> #memoryExpand(INDEX, _) ... </k>
-         <localMem> LM (.Map => INDEX |-> .Bytes) </localMem>
+    rule <k> #memoryExpand(INDEX, NEWSIZE) => #deductMemory(PEAK) ... </k>
+         <localMem> LM </localMem>
+         <currentMemory> CURR => CURR +Int maxInt(0, NEWSIZE -Int bytesInWords(0)) </currentMemory>
```
Comment (bmmoore) [inline iele-gas.md]: What is the maxInt for? Isn't `bytesInWords(0)` always `0`? Is memoryExpand ever used with NEWSIZE negative?

File: iele.md [hunk 3]
```diff
@@ -754,22 +754,35 @@ These operations are getters/setters of the local execution memory.
...
+
+    rule <k> #exec store _ , CELL , _ , 0 => . ... </k>
+         <localMem> LM </localMem>
       requires notBool CELL in_keys(LM)
 
     rule <k> #exec store VALUE , CELL => . ... </k>
          <localMem> LM => LM [ CELL <- Int2Bytes(VALUE, LE, Signed) ] </localMem>
+      requires CELL in_keys(LM) orBool VALUE =/=Int 0
+
+    rule <k> #exec store 0 , CELL => . ... </k>
```
Comment (bmmoore) [inline iele.md]: Would it be better to have one rule that always calls a function that decides whether or not to actually insert a map item?

File: data.md [hunk 4]
```diff
@@ -501,7 +501,7 @@ Encoding
     syntax String ::= #rlpEncodeWord ( Int )            [function]
                     | #rlpEncodeBytes ( Int , Int )     [function]
                     | #rlpEncodeString ( String )       [function]
-                    | #rlpEncodeInts ( Ints ) [function]
+                    | #rlpEncodeInts ( Ints ) [function, klabel(rlpEncodeInts), symbol]
```
[the remark this answers is not in the document]
Comment (dwightguth) [reply data.md]: This is assigning an explicit name so it can be used by the vm server C++ code.

File: data.md (hunk 1 above)
Comment (dwightguth) [reply data.md]: Same here.

File: data.md [hunk 5]
```diff
@@ -155,12 +155,14 @@ You could alternatively calculate `I1 %Int I2`, then add one to the normal integ
...
+                 | intSizes ( Array , Int , Int , Schedule ) [function, klabel(intSizesAux)]
  // -----------------------------------------------------------------------------
-    rule intSizes(ARR::Array, I) => intSizes(ARR, I, 0)
-    rule intSizes(ARR::Array, I, I) => 0
-    rule intSizes(ARR, I, J) => getInt(ARR [ J ]) +Int intSizes(ARR, I, J +Int 1) [owise]
+    rule intSizes(ARR::Array, I, SCHED) => intSizes(ARR, I, 0, SCHED)
+    rule intSizes(ARR::Array, I, I, _) => 0
+    rule intSizes(ARR, I, J, SCHED) => intSize(getInt(ARR [ J ])) +Int intSizes(ARR, I, J +Int 1, SCHED)
+      requires SCHED =/=K ALBE [owise]
+    rule intSizes(ARR, I, J, ALBE) => getInt(ARR [ J ]) +Int intSizes(ARR, I, J +Int 1, ALBE) [owise]
```
[the remark this answers is not in the document]
Comment (dwightguth) [reply data.md]: Yes, the idea is that this will be faster.

**Pull request merged**